### DISCUSSION TITLE
Process Crashes and not firing CoreWebView2.ProcessFailed

### DISCUSSION THREAD

**Discussion by RickStrahl** [MicrosoftEdge/WebView2Feedback#3932] [Q&A]
In my application, I'm trapping the CoreWebView2.ProcessFailed event for all instances of the WebBrowser control. In the event handler I handle the event something like this:

```cs
private void CoreWebView2_ProcessFailed(object sender, CoreWebView2ProcessFailedEventArgs e)
{

    mmApp.Log($"Editor WebView Control Process Failed: {e.ProcessFailedKind} {e.Reason}",
        logLevel: LogLevels.Critical);
    
    if (Editor?.MarkdownDocument != null  &&
        !Editor.MarkdownDocument.IsUntitled &&
        Editor.MarkdownDocument.CurrentText != Editor.MarkdownDocument.OriginalText &&
        MessageBox.Show(mmApp.Model.Window,
            "The Editor WebView Control has failed and the application has to shut down. Please restart Markdown Monster.\n\n" +
            "Do you want to save your pending document changes in\n" +
            $"{Editor.MarkdownDocument.FilenameWithIndicator}?",
            "Shutdown and Save",
            MessageBoxButton.YesNo, MessageBoxImage.Error) == MessageBoxResult.Yes)
        {
            Editor.MarkdownDocument.Save();

            var window = mmApp.Model.Window;
            foreach (object ttab in window.TabControl.Items)
            {
                if (ttab is TabItem tab && tab.Tag == Editor)
                {
                    window.CloseTab(tab).FireAndForget();
                    return;
                }
            }
        }
    else
    {

        MessageBox.Show(mmApp.Model.Window,
            "The Editor WebView Control has failed and the application has to shut down. Please restart Markdown Monster.",
            "Shutdown Notice",
            MessageBoxButton.OK, MessageBoxImage.Error);
    }

    Environment.Exit(1);
}
```

In essence the application exits after the failure. There are several different WebViews in use - the one above is critcal if it doesn't work the browser can't be reset.

However in my logs I see long strings of these errors:

![image](https://user-images.githubusercontent.com/1374013/150884654-f9a3abdf-1263-42a6-9622-04805a1c5a6f.png)

but I don't see the Process_Failed errors  in my log (the `mmApp.Log()` call), only the exception when it hits which is logged at the process level exception handler.

### Other

This is anecdotal but the vast majority of these failures occur with Asian languages although that's besides the point of this post. But the crashes very heavily lean towards those languages.

**Comment by champnic** [comment]
Hey @RickStrahl - thanks for this issue. I took a quick look at VerifyBrowserNotCrashed, and the _browserCrashed variable is getting set when the WPF control listens to ProcessFailed. So ProcessFailed is working in general when the browser crashes. When the WPF control gets ProcessFailed it will also Uninitialize the control - I'm wondering if that or something else might somehow trigger the UpdateLayout which calls the Verify check before the ProcessFailed handler in you app has a chance to run?

**Comment by RickStrahl** [comment]
I don't see this failure myself, I just see it pop up somewhat frequently in my user logs. 

I've added some additional logging to try and figure out exactly which control is failing, so hopefully this will get me a little close to see what the problem is. I have a feeling that you're right - there's some sort of recursive thing happening where the control is uninitialized and it's trying to recreate itself and then fails again - many of the errors do show up as a clump of requests...

Will let you know what I find once I let the logs run for a few days of the latest update.
